pkg/controller/noderesource/node_resource_controller.go

```diff
@@ -0,0 +1,171 @@
+		// TODO fix: strategic merge patch kubernetes
+		if err := r.Client.Status().Update(context.TODO(), nodeCopy); err != nil {
+			r.Recorder.Event(tsp, v1.EventTypeNormal, "FailedUpdateNodeExtendResource", err.Error())
+			r.Log.Error(err, "update Node status extend-resource error: %v", err)
+			return ctrl.Result{}, err
+		}
+		r.Recorder.Event(tsp, v1.EventTypeNormal, "UpdateNode", "Update Node Extend Resource Success")
+	}
+	return ctrl.Result{}, nil
+}
+
+func (r *NodeResourceReconciler) FindTargetNode(ctx context.Context, tsp *predictionapi.TimeSeriesPrediction) (*v1.Node, error, bool) {
+	address := tsp.Spec.TargetRef.Name
+	if address == "" {
+		return nil, fmt.Errorf("target is not specified"), false
+	}
+	nodeList := &v1.NodeList{}
+	if err := r.Client.List(ctx, nodeList); err != nil {
+		r.Log.Error(err, "list node error")
+		return nil, err, true
+	}
+	// the reason we use node ip instead of node name as the target name is
+	// some monitoring system does not persist node name
+	for _, n := range nodeList.Items {
+		for _, addr := range n.Status.Addresses {
+			if addr.Address == address {
+				r.Log.Info("Found target node", "nodeName", n.Name)
+				return &n, nil, false
+			}
+		}
+	}
+	return nil, fmt.Errorf("target [%s] not found", address), false
+}
+
+func (r *NodeResourceReconciler) BuildNodeStatus(tsp *predictionapi.TimeSeriesPrediction, node *v1.Node) {
+	idToResourceMap := map[string]*v1.ResourceName{}
+	for _, metrics := range tsp.Spec.PredictionMetrics {
+		if metrics.ResourceQuery == nil {
+			continue
+		}
+		idToResourceMap[metrics.ResourceIdentifier] = metrics.ResourceQuery
+	}
+	// build node status
+	nextPredictionResourceStatus := &tsp.Status
+	for _, metrics := range nextPredictionResourceStatus.PredictionMetrics {
+		resourceName, exists := idToResourceMap[metrics.ResourceIdentifier]
+		if !exists {
+			continue
+		}
+		for _, timeSeries := range metrics.Prediction {
+			var maxUsage, nextUsage int64
+			var nextUsageFloat float64
+			var err error
+			for _, sample := range timeSeries.Samples {
+				if nextUsageFloat, err = strconv.ParseFloat(sample.Value, 64); err != nil {
+					r.Log.Error(err, "parse extend resource value error, resource value: %s", sample.Value)
+					continue
+				}
+				nextUsage = int64(nextUsageFloat)
+				if maxUsage < nextUsage {
+					maxUsage = nextUsage
+				}
+			}
+			var nextRecommendation int64
+			switch *resourceName {
+			case v1.ResourceCPU:
+				nextRecommendation = node.Status.Allocatable.Cpu().Value() - maxUsage
+			case v1.ResourceMemory:
+				nextRecommendation = node.Status.Allocatable.Memory().Value() - maxUsage
+			default:
+				continue
+			}
+			extResourceName := fmt.Sprintf(ExtResourcePrefix, string(*resourceName))
+			resValue, exists := node.Status.Capacity[v1.ResourceName(extResourceName)]
+			if exists && float64(resValue.Value())/float64(nextRecommendation) <= 1+MinDeltaRatio {
+				continue
+			}
+			node.Status.Capacity[v1.ResourceName(extResourceName)] =
```

Comment:
The scheduler use node.Status.Allocatable to do scheduling,  we should update the node.Status.Allocatable too.

Reply:
done